File: client-module-server/src/main/java/br/edu/les/module/client/domain/Estado.java
package br.edu.les.module.client.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;

import javax.persistence.*;

@Component
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "estado")
public class Estado extends EntidadeDominio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "est_id")
    private Integer id;

    @Column(name = "est_descricao")
    private String descricao;

}

File: client-module-server-node/src/models/Estado.ts
import { Column, Entity, OneToMany, OneToOne, PrimaryGeneratedColumn } from "typeorm"
import { Cidade } from "./Cidade";

@Entity("estado")
class Estado {
    
    @PrimaryGeneratedColumn({name: "est_id"})
    id: number;

    @Column({name: "est_descricao"})
    descricao: string;

    @OneToMany(type => Cidade, cidades => cidades.estado)
    cidades: Cidade[];
}

export{ Estado }